app/models/user.rb

```diff
@@ -5,4 +5,14 @@ class User < ApplicationRecord
   has_many :books, through: :checkouts
 
   validates :name, presence: true
+
+  def total_fines
+    total_fine_amount = 0
+    self.checkouts.each do |checkout|
```

Comment:
The combination of "initialize a value to some beginning value then use `each` to mutate that value for each value in a collection" is a bit of an anti-pattern because it's covered by other methods in Ruby. In this case, you could use either `reduce` (also called `inject` -- they are the same method) or `each_with_object`.

Both of these make it so you don't need to do the "initialize some value, mutate it inside the block, then return the mutated value" dance. `each_with_object` is preferred by most these days (though I still have a habit of using `reduce`) because it has 1 less gotcha, which you can read about here: http://technology.customink.com/blog/2014/10/14/better-hash-injection-using-each-with-object/

I'd encourage you to read up the docs for these methods and try rewriting with `reduce` and `each_with_object`. If you get stuck, I can help you out.